src/commands/subscription.ts

```diff
@@ -423,19 +448,21 @@ export function getListeners(
     _league: league.League,
     sources: string[],
     event: string
-): Promise<string[]> {
+): Promise<TargetWithSourceContext[]> {
     return new Promise((resolve, reject) => {
-        sources = _.map(sources, _.toLower)
+        const playerNamesWithContext: SourceWithContext[] = sources
+            .map((source, index) => [source.toLowerCase(), SOURCE_CONTEXT_ARRAY[index]] as [string, SourceContext])
+            .map(transformZippedSourceContext)
         // These are names of users, not users. So we have to go get the real
         // user and teams. This is somewhat wasteful - but I'm not sure whether
         // this is the right design or if we should pass in users to this point.
-        const teamNames = _(sources)
-            .map((s) => _league.getTeamByPlayerName(s))
-            .filter(isDefined)
-            .map('name')
-            .map(_.toLower)
-            .value()
-        const possibleSources = _.concat(sources, teamNames)
+        const teamNamesWithContext: SourceWithContext[] = sources
```

Comment:
This was my first crack at a data model for passing around the source context - could potentially be extended to add whether notification came from a team (which could help solve #109). This is definitely the biggest change here.